WordPress/src/main/java/org/wordpress/android/ui/comments/CommentDetailFragment.java

```diff
@@ -282,7 +284,31 @@ public void onClick(View v) {
         mBtnTrashComment.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                moderateComment(CommentStatus.TRASH);
+                if (mComment.getStatusEnum() == CommentStatus.TRASH) {
+                    AlertDialog.Builder dialogBuilder = new AlertDialog.Builder(
+                            getActivity());
+                    dialogBuilder.setTitle(getResources().getText(R.string.trash));
+                    dialogBuilder.setMessage(getResources().getText(R.string.dlg_sure_to_delete_comment));
+                    dialogBuilder.setPositiveButton(getResources().getText(R.string.yes),
+                            new DialogInterface.OnClickListener() {
+                                public void onClick(DialogInterface dialog, int whichButton) {
+                                    moderateComment(CommentStatus.DELETE);
+                                }
+                            });
+                    dialogBuilder.setNegativeButton(
+                            getResources().getText(R.string.no),
+                            new DialogInterface.OnClickListener() {
```

Comment:
Just an FYI: you can pass `null` here since the click listener does nothing (something I didn't know until relatively recently).